src/util/Constants.js

```diff
@@ -14,68 +14,50 @@ exports.WSCodes = {
   4014: 'DISALLOWED_INTENTS',
 };
 
-const AllowedImageFormats = ['webp', 'png', 'jpg', 'jpeg', 'gif'];
+const AllowedImageFormats = ['webp', 'png', 'jpg', 'jpeg'];
 
 const AllowedImageSizes = [16, 32, 56, 64, 96, 128, 256, 300, 512, 600, 1024, 2048, 4096];
 
-function makeImageUrl(root, { format = 'webp', size } = {}) {
+function makeImageUrl(root, { hash, format = 'webp', forceStatic = false, size } = {}) {
   if (!['undefined', 'number'].includes(typeof size)) throw new TypeError('INVALID_TYPE', 'size', 'number');
-  if (format && !AllowedImageFormats.includes(format)) throw new Error('IMAGE_FORMAT', format);
+  if (!AllowedImageFormats.includes(format)) throw new Error('IMAGE_FORMAT', format);
   if (size && !AllowedImageSizes.includes(size)) throw new RangeError('IMAGE_SIZE', size);
-  return `${root}.${format}${size ? `?size=${size}` : ''}`;
+  if (forceStatic) format ??= 'webp';
+  else if (hash?.startsWith('a_')) format = 'gif';
+  return `${root}${hash ? `/${hash}` : ''}.${format}${size ? `?size=${size}` : ''}`;
 }
 
 /**
- * Options for Image URLs.
- * @typedef {StaticImageURLOptions} ImageURLOptions
- * @property {boolean} [dynamic=false] If true, the format will dynamically change to `gif` for animated avatars.
- */
-
-/**
- * Options for static Image URLs.
- * @typedef {Object} StaticImageURLOptions
- * @property {string} [format='webp'] One of `webp`, `png`, `jpg`, `jpeg`.
+ * Options for image URLs.
+ * @typedef {Object} ImageURLOptions
+ * @property {string} [format] One of `webp`, `png`, `jpg`, `jpeg`.
+ * @property {boolean} [forceStatic=false] If `true`, the format will be as specified or `webp`.
```

Comment:
Why can’t this stay as dynamic and default to true? People could set it to false if they want the static image

Reply:
imo a default-false boolean is better than default true. though i think `force` is unnecessary and it can be called just `static`

Reply:
`static` is a reserved keyword.

Reply:
Why not using `steady`? It's a smaller word with the same meaning.